clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10218,10 +10218,23 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
     bool IsCPU = Triple.isSPIR() &&
                  Triple.getSubArch() == llvm::Triple::SPIRSubArch_x86_64;
     TranslatorArgs.push_back("-spirv-max-version=1.4");
-    TranslatorArgs.push_back("-spirv-debug-info-version=ocl-100");
-    // Prevent crash in the translator if input IR contains DIExpression
-    // operations which don't have mapping to OpenCL.DebugInfo.100 spec.
-    TranslatorArgs.push_back("-spirv-allow-extra-diexpressions");
+    // Enable NonSemanticShaderDebugInfo.200 for CPU AOT and for non-Windows
+    // JIT. Don't enable on FPGA H/W.
+    const llvm::Triple *AuxTriple = TC.getAuxTriple();
+    const bool IsWindowsMSVC =
+        Triple.isWindowsMSVCEnvironment() ||
+        (AuxTriple && AuxTriple->isWindowsMSVCEnvironment());
+    const bool EnableNonSemanticDebug =
+        IsCPU || (!IsWindowsMSVC && !C.getDriver().IsFPGAHWMode());
```

Comment:
Comment states 'non-Windows JIT'.  This check allows for all non-windows, including AOT.  Is that OK?

Reply:
Apologies to the long response. Yes, non-Windows AOT is also OK until it's not FPGA.